engineering/conventions-go.md

```diff
@@ -0,0 +1,50 @@
+
+# Go Development Conventions
+
+## Introduction
+
+This guide documents development conventions for Go at source{d}. Check [general conventions](conventions.md) for language-independent conventions that are also applicable for Go projects.
+
+## Supported Go Versions
+
+* Support latest two stable major versions of Go (e.g. 1.9.x and 1.10.x).
+  * Some of our projects may support more versions, specially when there is wide-adoption outside source{d} (e.g. [go-git](https://github.com/src-d/go-git) currently supports three).
+
+## Dependency Management
+
+* If your project is an application:
+  * Use [dep](https://github.com/golang/dep) to manage dependencies.
+  * Check in the `vendor` directory into git.
+* If your project is a library:
+  * You may use [dep](https://github.com/golang/dep) to manage dependencyes if needed. If you do, do not commit `vendor` directory.
+
+## Build System
+
+* If your project requires fetching or building dependencies, implement that in the `dependencies` rule.
+* If binaries can be built, do so in the `packages` rule. Note that [src-d/ci](https://github.com/src-d/ci) provides that rule by default.
+
+## Testing
+
+* Use [testify](https://github.com/stretchr/testify).
+* Use [suites](https://github.com/stretchr/testify#suite-package) to share setup/teardown code among related tests. When testing multiple methods of the same struct, you generally want to start with a suite.
+
+## Executables
+
+* Put sources for your executable commands in `cmd/<your command>`.
```

Comment:
@mcuadros also mentioned that main files should be called `main.go`. Some projects call it with the name of the command, so it's worth mentioning.

Reply:
Yes, this is not correct, the commands should be in folder witt the name of the command and the file, the most standard is be called `main.go`.

So the rule can be resumed in `cmd/<command-name>/main.go`

Otherwise when you make `go install` your binary is going to be called `cmd`